Drop down overhaul

girka2k opened the pull request.

Alewx committed:
Merge pull request #6 from girka2k/dev

regular update

Alewx committed:
Merge pull request #7 from Alewx/Release

Release Update changes

Alewx committed:
Merge pull request #8 from girka2k/dev

regular update

Alewx committed:
Initial Dropdown addition

- first try to update the dropdown menus
- used the stuff from the kerbal alarmclock as they looked rocksolid
- so far no solution more advanced than imagined

Alewx committed:
First actual working version

- TriggerDropdown as temporary base
- AdvancedDropdown is an updated version of the dropdowns
- Dropdownmanager managed events of the Dropdowns
- Added central getters for the guistlye and guiskin

Alewx committed:
hopefully merged development into dropdownoverhaul

merge of development

Alewx committed:
working

-slight codestyle changes
-adapted window to make the dropdown work
-removed window text to reduce confusion

Alewx commented:
it is only a testing version with working debug outputs for the different values; guistyle is not correct due to current use of extermal class.
feel free to work on it, I will also try to make further progress.

Alewx committed:
Dropdown overhaul complete

-deleted the unneeded classes from alarm clock
-changed the function to draw the info window, slightly improved to work
with rect array
-working dropdowns that close on outside clicks

Alewx commented:
ready for merging.
